versioned_docs/version-4.0.0/tributech_node/Webhook_integration.md

```diff
@@ -0,0 +1,230 @@
+
+Its important to note that each of the Event Types contain a predefined property `EventQoS` which will be contained in each
+webhook event indicating the frequency of the events. This property is used by the redelivery mechanism to know how often a event should be handled in case 
+the HTTP POST client response with an error code (more details in [Error Handling](#error-handling)). 
+
+#### Device Events
+Device Events are triggered by interactions with devices and device status changes,
+this includes enrollments, configuration and source commands.
+
+|Name | Description |
+|--|--|
+|DeviceConnectionEvent|Occurs when a Device has changed its connection status|
+|DeviceStateEvent|Occurs when a Device has changed its own state|
+|GetConfigurationRequestEvent|Occurs when the Get Configuration Command has been requested|
+|GetConfigurationResponseEvent|Occurs when the Get Configuration Command has finished and the response was received|
+|IncomingEnrollmentRequestEvent|Occurs when an Enrollment request was received|
+|SetConfigurationRequestEvent|Occurs when the Set Configuration Command has been requested|
+|SetConfigurationResponseEvent|Occurs when the Set Configuration Command has finished and the response was received|
+|SourceCommandRequestEvent|Occurs when a Source Command is invoked and sent to the Agent|
+|SourceCommandResponseEvent|Occurs when the Source Command has finished and the response was received|
+|SourceCommandStateUpdateEvent|Occurs when a Agent or Agent Source has sent a execution state of the current Command|
+
+#### Proof Events
+Proof Events describe events related to proof handling and status, e.g. received, valid, stored.
+
+|Name | Description |
+|--|--|
+|ProofConfirmedEvent|Occurs when the Proof is successfully placed on the BlockChain|
+|ProofReceivedEvent|Occurs when a Proof is received from a Device|
+|ProofStoredEvent|Occurs when the Proof is stored into the Persistence Layer and passed into the Block Chain|
+|ProofValidatedEvent|Occurs when the Proof is validated|
+
+#### Stream Events
+Stream Events provide information when stream management operations are executed.
+
+|Name | Description |
+|--|--|
+|StreamDeletedEvent|Occurs when a Agent or Stream is deleted and will be removed from the Node|
+
+#### Twin Events
+Twin Events provide information when Digital Twin management operations are executed.
+
+|Name | Description |
+|--|--|
+|TwinInstanceDeletedEvent|Occurs when a Digital Twin Instance was removed from the Node|
+|TwinModelDeletedEvent|Occurs when a Digital Twin Model was removed from the Node|
+
+#### Value Events
+Value Events are triggered when the Tributech Node interacts with Stream Values.
+
+|Name | Description |
+|--|--|
+|ValueReceivedEvent|Occurs when a Value is received from an Device|
+|ValueStoredEvent|Occurs when a Value is stored into the Persistence Layer|
+
+#### Webhook Events
```

Comment:
This is the general Webhook Event and has no functional use (base abstract class) thats an issue on demeter showing events which should not be shown. But remove it for now from the docs please